src/hazelcore/semantics/CursorInfo.re

```diff
@@ -985,25 +1197,31 @@ and _syn_cursor_info_line =
       ctx: Contexts.t,
       zli: ZExp.zline,
     )
-    : option(t) =>
+    : option(deferrable_t) =>
   switch (zli) {
   | CursorL(cursor, line) =>
     let (prefix, _, suffix) = frame;
     Some(
-      mk_cursor_info(
-        ~subblock_start=line_no,
-        OnLine,
-        Line(line),
-        ExpFrame(prefix, None, suffix),
-        cursor,
-        ctx,
-        node_steps,
-        term_steps,
+      CursorNotOnDeferredVarPat(
+        mk_cursor_info(
+          ~subblock_start=line_no,
+          OnLine,
+          Line(line),
+          ExpFrame(prefix, None, suffix),
+          cursor,
+          ctx,
+          node_steps,
+          term_steps,
+        ),
       ),
     );
   | ExpLineZ(ze) =>
-    _syn_cursor_info(~node_steps, ~term_steps, ~frame, ctx, ze)
+    switch (_syn_cursor_info(~node_steps, ~term_steps, ~frame, ctx, ze)) {
+    | None => None
+    | Some(ci) => Some(CursorNotOnDeferredVarPat(ci))
+    }
   | LetLineZP(zp, ann, block) =>
```

Comment:
isn't this the case where we need to consider whether it was deferred? am I not seeing the code?

Reply:
I feel like the only case needs CursordefferedVarPat is when the cursor is on LetLineZP. The only two cases that ZExp.t involves ZPat.t is LamZP and CaseZR. And, outside of ZExp.t, any variable defined in the ZExp.t should have already been out of scope. Please correct me if I miss anything.

Reply:
oh nevermind, I see how this works